src/java.base/share/classes/java/io/SequenceInputStream.java

```diff
@@ -238,4 +238,19 @@ public void close() throws IOException {
             throw ioe;
         }
     }
+
+    @Override
+    public long transferTo(OutputStream out) throws IOException {
+        Objects.requireNonNull(out, "out");
+        if (getClass() == SequenceInputStream.class) {
```

Comment:
For BIS we had to restrict optimizations to direct usages because that class exposes its internals to subclasses via protected fields. We don't have this with SIS. On the other hand, SIS is non-final, dates from JDK 1.0, so we have to assume there are "transferTo unaware" subclasses that override the read methods to intercept all access. So I think being conservative and checking for subclassing is right. It doesn't prevent it being relaxed in the future.

Reply:
Exactly what I had in mind when authoring this code change. :-)